dexmaker/src/main/java/com/google/dexmaker/DexMaker.java

```diff
@@ -323,6 +327,50 @@ public void declare(FieldId<?, ?> fieldId, int flags, Object staticValue) {
         }
     }
 
+    // Generate a file name for the jar by taking a checksum of MethodIds and
+    // parent class types.
+    private String generateFileName() {
+        int checksum = 1;
+
+        Set<TypeId<?>> typesKeySet = types.keySet();
+        Iterator<TypeId<?>> it = typesKeySet.iterator();
+        int[] checksums = new int[typesKeySet.size()];
+        int i = 0;
+
+        while (it.hasNext()) {
+            TypeId<?> typeId = it.next();
+            TypeDeclaration decl = getTypeDeclaration(typeId);
+            Set<MethodId> methodSet = decl.methods.keySet();
+            checksums[i++] = 31 * decl.supertype.hashCode() + methodSet.hashCode();
```

Comment:
I believe you told me the other day that the methodSet is the union of the class methods, the parent class methods on up the chain, correct?

Should we include the FieldDeclaration in the hash as well?  It seems like "no", but figured I'd ask.